x/wasm-storage/keeper/filter.go

```diff
@@ -39,3 +54,54 @@ func ApplyFilter(filter []byte, reveals []RevealBody) ([]int, bool, error) {
 		return outliers, false, nil
 	}
 }
+
+// filterMode takes in a list of reveals, and a json path to get the
+// filtering data from the RevealBody->reveal.
+// The function returns a list of int where value of each element can be
+// either 0 or 1. At index i Value 1 means RevealBody[i] contains an
+// outlier. If a reveal have MORE THAN 2/3 frequency, than it's not an
+// outlier.
+//
+// The function also returns if a consensus is reached. Consensus can be
+// reached in two ways.
+// 1. More than 1/3 of the reveals are corrupted. i.e. has non-zero exit code.
+// 2. More than 2/3 of the data is the same.
+func filterMode(dataPath string, reveals []RevealBody) ([]int, bool) {
+	var maxFreq, nonZeroExitCode int
+	freq := make(map[string]int, len(reveals))
+	revealVals := make([]string, 0, len(reveals))
+
+	for _, r := range reveals {
+		val := gjson.Get(r.Reveal, dataPath).String()
+		revealVals = append(revealVals, val)
+		if r.ExitCode != 0 {
+			nonZeroExitCode++
+		}
+		freq[val]++
+		maxFreq = max(freq[val], maxFreq)
+	}
+
+	outliers := make([]int, len(reveals))
+	for i := 0; i < len(outliers); i++ {
+		outliers[i] = 1
+	}
+
+	// If more than 1/3 reveals have non exit code, we reached a consensus that
+	// these are unusable reveals.
+	if nonZeroExitCode*3 > len(reveals) {
+		return outliers, true
+	}
+
+	// If not MORE THAN 2/3 matches the max frequency, there is no data-consensus.
+	if maxFreq*3 <= len(reveals)*2 {
+		return outliers, false
+	}
```

Comment:
I think this should be strictly `<`, so that if we have replication factor 3, then with 1 outlier, there will be consensus. Wdyt?
```suggestion
	// If not MORE THAN 2/3 matches the max frequency, there is no data-consensus.
	if maxFreq*3 < len(reveals)*2 {
		return outliers, false
	}
```

Reply:
Now consensus reaches if **at lease** 2/3 is matched.
Before it was **more than** 2/3

[9bae247](https://github.com/sedaprotocol/seda-chain/pull/290/commits/9bae247a8adb32061b10bf4ddc58775c1260ee6c)